integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/receipt/ReceiptStrategyFactory.java

```diff
@@ -0,0 +1,37 @@
+package no.difi.meldingsutveksling.receipt;
+
+import no.difi.meldingsutveksling.receipt.strategy.DpiReceiptStrategy;
+import no.difi.meldingsutveksling.receipt.strategy.DpvReceiptStrategy;
+import no.difi.meldingsutveksling.receipt.strategy.EduReceiptStrategy;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ReceiptStrategyFactory {
+
+    private static DpiReceiptStrategy dpiReceiptStrategy;
+    private static DpvReceiptStrategy dpvReceiptStrategy;
+    private static EduReceiptStrategy eduReceiptStrategy;
+
+    @Autowired
+    ReceiptStrategyFactory(DpiReceiptStrategy dpiReceiptStrategy,
+                           DpvReceiptStrategy dpvReceiptStrategy,
+                           EduReceiptStrategy eduReceiptStrategy) {
+        this.dpiReceiptStrategy = dpiReceiptStrategy;
+        this.dpvReceiptStrategy = dpvReceiptStrategy;
+        this.eduReceiptStrategy = eduReceiptStrategy;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this instance-reference to a static reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS2209)